crates/rolldown_common/src/types/symbol_ref.rs

```diff
@@ -14,3 +15,55 @@ impl From<(NormalModuleId, SymbolId)> for SymbolRef {
     Self { owner: value.0, symbol: value.1 }
   }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum SymbolOrMemberExprRef {
+  Symbol(SymbolRef),
+  MemberExpr(MemberExprRef),
+}
+
+impl SymbolOrMemberExprRef {
+  /// get the first part of the expression,
+  /// e.g. `test.a.b` will return `test`
+  /// for identifier, it will return itself
+  pub fn first_part(&self) -> &SymbolRef {
```

Comment:
```suggestion
  pub fn symbol_ref(&self) -> &SymbolRef {
```

I think it's ok you just call it `symbol` or `symbol_ref`. Because when we are referring `symbol` in `test.a.b`, the symbol would still be `test`. `.a` and `.b` might point to some symbols, but for the expression `test.a.b` itself, the symbol would only be `test`.